test/fixtures/gcs-fixture/src/main/java/fixture/gcs/GoogleCloudStorageHttpHandler.java

```diff
@@ -237,14 +237,16 @@ public void handle(final HttpExchange exchange) throws IOException {
                 BytesReference blob = blobs.get(blobName);
                 final String range = exchange.getRequestHeaders().getFirst("Content-Range");
                 final Integer limit = getContentRangeLimit(range);
-                final int start = getContentRangeStart(range);
-                final int end = getContentRangeEnd(range);
 
                 blob = CompositeBytesReference.of(blob, requestBody);
                 blobs.put(blobName, blob);
 
                 if (limit == null) {
-                    exchange.getResponseHeaders().add("Range", String.format(Locale.ROOT, "bytes=%d/%d", start, end));
+                    if ("bytes */*".equals(range) == false) {
+                        final int start = getContentRangeStart(range);
+                        final int end = getContentRangeEnd(range);
+                        exchange.getResponseHeaders().add("Range", String.format(Locale.ROOT, "bytes=%d-%d", start, end));
```

Comment:
This may actually fix the JDK-8 issue we've been seeing (I'm not 100% on this yet). The header format was broken here and the retries tests with the new SDK caught this now. I wonder if this was the spot that made the JDK-8 URL client spontaneously close connections on header parse failures (as it turns out it doesn't log/thrown on those in any way). I'll revisit this in a follow-up on 7.x.

Reply:
Good catch!

Reply:
Exception handling in `HttpServer` is pretty debugging-unfriendly, see https://github.com/elastic/elasticsearch/issues/68967.